src/common/http/http.cpp

```diff
@@ -146,7 +146,13 @@ error::Error BreakDownUrl(const string &url, BrokenDownUrl &address, bool with_a
 			address = {};
 			return error::Error(port.error().code, url + " contains invalid port number");
 		}
-		address.port = port.value();
+		if (port.value() < 0 or port.value() > numeric_limits<decltype(address.port)>::max()) {
+			address = {};
+			return error::Error(
+				make_error_condition(errc::invalid_argument),
+				url + " contains invalid port number");
+		}
+		address.port = static_cast<decltype(address.port)>(port.value());
```

Comment:
Nice. We haven't used `decltype` but we should have.

Reply:
Yes, it's nice.

But now I see this contradicts my answer in the previous comment. Do I need a cast here now...? Let me recheck.

Reply:
Following my other comment above, this is no longer a cast, but I still use `decltype` in the `StringTo` call, which then does range checking based on that type. So I removed the explicit range checking from here, and just rely on the return value.